### PR TITLE
added guard for idle stream shutdown

### PR DESCRIPTION
**Pull request by JamesJahraus**
My server sent </stream:stream> stanza because I am not replying to presence etc.

I created a guard to catch the </stream:stream> stanza when it arrives, and I use the existing function in client handle_info to shut down the stream.

Researched:
http://xmpp.org/extensions/xep-0190.html

### CODE REVIEW

File: lib/hedwig/transports/tcp.ex [hunk 1]
```diff
@@ -163,6 +164,14 @@ defmodule Hedwig.Transports.TCP do
     {:noreply, new_state}
   end
 
+  @doc """
+  Gaurd for closing idle stream. xmpp.org XEP-0190
+  """
+  defp handle_data(_socket, data, state) when data == "</stream:stream>" do
+    Logger.warn fn -> "Incoming close idle stream: #{inspect data}" end
+    Client.handle_info({:stop, "Idle stream shutdown."}, Client.get(state.client))
```
Comment (scrogson) [inline lib/hedwig/transports/tcp.ex]: `handle_info` is an OTP callback that is invoked by send a process a message. It is not meant to be called directly.

File: lib/hedwig/transports/tcp.ex [hunk 2]
```diff
@@ -153,7 +154,7 @@ defmodule Hedwig.Transports.TCP do
     {:noreply, state}
   end
 
-  defp handle_data(_socket, data, state) do
+  defp handle_data(_socket, data, state) when data != "</stream:stream>" do
```
Comment (scrogson) [inline lib/hedwig/transports/tcp.ex]: I think I would prefer that this be handled within a `case` of the parsed data rather than a guard.

**Comment by scrogson** [comment]
Hi @JamesJahraus,

I've made some comments. What server are you using?

**Comment by JamesJahraus** [comment]
Thanks @scrogson I will give this a try.  The server is an openfire sever, a component in our stack.  It basically communicates to android endpoints.  I am trying to simulate the endpoints, for testing, using hedwig.

**Comment by scrogson** [comment]
Hi @JamesJahraus,

I'm going to go ahead and close this PR as I won't be merging it. My suggestion would be to write a small handler that responds to the presence messages that are causing your server to close your connection.

If you need help with that, please reach out.

**Pull request closed without merging**